Client/Frontend/Home/Wallpapers/v2/NetworkServices/WallpaperNetworkModule.swift

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+class WallpaperNetworkingModule: WallpaperNetworking {
+
+    /// A basic async/await wrapper
+    func data(from url: URL) async throws -> (Data, URLResponse) {
+        return try await withCheckedThrowingContinuation { continuation in
+            URLSession.shared.dataTask(with: url) { data, response, error in
```

Comment:
Making the [comment again](https://github.com/mozilla-mobile/firefox-ios/pull/11400#discussion_r930065233): 
```
URLSession.shared cannot be mocked or a custom url session cannot be user 🤔 Hence WallpaperNetworkingModule cannot be tested at the moment?
```

Reply:
I'm not sure I understand what you mean by this. Are you asking why I haven't tested this? I haven't because I guess the question is, do we test URLSession.shared reaching out to the internet?

I have tested our side using `NetworkingMock` and adapting it to the same protocol `WallpaperNetworking`. But if that's not what you mean, I'm confused.

Reply:
Discussed with Roux, but basically testing the WallpaperNetworkModule code by using a mock URLSession